### PR TITLE
React query setup

### PR DESCRIPTION
**Pull request by Loges2108**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -42,6 +47,8 @@ function App() {
         </Routes>
       </BrowserRouter>
     </ThemeProvider>
+    <ReactQueryDevtools initialIsOpen buttonPosition="bottom-left" />
```
Comment (sathyaprabaS) [inline src/App.tsx]: remove devTools

File: src/common/component/commonpages/CommonPage.tsx [hunk 2]
```diff
@@ -12,36 +12,41 @@ import FormControl from "@mui/material/FormControl";
...
-  };
 
   const handleExpandClick = () => {
     setExpandDescription(!expandDescription);
   };
 
-  const handleChange = (event: SelectChangeEvent) => {
-    setSortProductOption(event.target.value as string);
+  const handleSortChange = (event: SelectChangeEvent) => {
+    setSortProductOption(event.target.value as SortingOption);
```
Comment (sathyaprabaS) [inline src/common/component/commonpages/CommonPage.tsx]: why sorting option give  as event use properly

File: src/common/component/commonpages/CommonPage.tsx [hunk 3]
```diff
@@ -12,36 +12,41 @@ import FormControl from "@mui/material/FormControl";
...
 import MenuItem from "@mui/material/MenuItem";
 import CommonProductCard from "../commonCard/CommonProductCard";
+import { SortingOption, SortingOptionLabel, sortingOptions } from "../../../interface/type";
 
-interface Props {
-  collection: Collection;
-}
+const CommonPage = (props: Icommonpage) => {
+  const jewelleryItemWithCollection = props;
+  console.log(props);
```
Comment (sathyaprabaS) [inline src/common/component/commonpages/CommonPage.tsx]: remove console

File: src/common/component/commonpages/CommonPage.tsx (hunk 2 above)
Comment (sathyaprabaS) [inline src/common/component/commonpages/CommonPage.tsx]: how to identify interface mention I

File: src/common/component/commonpages/CommonPage.tsx [hunk 4]
```diff
@@ -12,36 +12,41 @@ import FormControl from "@mui/material/FormControl";
...
   };
 
+  const sortProducts = (products: Product[]): Product[] => {
+    switch (sortProductOption) {
+      case SortingOption.PriceLowToHigh:
+        return products.slice().sort((a, b) => a.price - b.price);
+      case SortingOption.PriceHighToLow:
+        return products.slice().sort((a, b) => b.price - a.price);
+      case SortingOption.NameAZ:
+        return products.slice().sort((a, b) => a.title.localeCompare(b.title));
```
Comment (sathyaprabaS) [inline src/common/component/commonpages/CommonPage.tsx]: name A to Z not working

File: src/common/component/commonpages/CommonPage.tsx [hunk 5]
```diff
@@ -12,36 +12,41 @@ import FormControl from "@mui/material/FormControl";
...
+      case SortingOption.NameZA:
```
Comment (sathyaprabaS) [inline src/common/component/commonpages/CommonPage.tsx]: name Z to A not working

File: src/interface/type.ts [hunk 6]
```diff
@@ -30,8 +37,37 @@ export interface ISignUp {
...
+  description?:string;
+  products?: Product[];
+}
+
+
+
+
+// sorting products
+
+export enum SortingOption {
```
Comment (sathyaprabaS) [inline src/interface/type.ts]: why use this page use enum page

File: src/pages/bracelets/Bracelates.tsx [hunk 7]
```diff
@@ -1,14 +1,30 @@
...
-  );
-
-  if (!braceletsCollection) return null;
+const Bracelets = () => {
+  const {
+    data: BraceletsCollection,
+    isLoading,
+    isError,
+  } = useGetAllItemsByCollectionName("Bracelets");
+  console.log(BraceletsCollection);
```
Comment (sathyaprabaS) [inline src/pages/bracelets/Bracelates.tsx]: remove console

File: src/common/component/commonCard/CommonProductCard.tsx [hunk 8]
```diff
@@ -4,18 +4,26 @@ import CardMedia from "@mui/material/CardMedia";
 import Typography from "@mui/material/Typography";
 import { Product } from "../../../interface/type";
 import { Link } from "react-router-dom";
+import { useState } from "react";
 
 
 
 interface Props {
```
Comment (sathyaprabaS) [inline src/common/component/commonCard/CommonProductCard.tsx:9]: use I for Inteface

File: src/App.tsx (hunk 1 above)
Comment (Loges2108) [reply src/App.tsx]: removed

File: src/common/component/commonpages/CommonPage.tsx (hunk 3 above)
Comment (Loges2108) [reply src/common/component/commonpages/CommonPage.tsx]: removed

File: src/pages/bracelets/Bracelates.tsx (hunk 7 above)
Comment (Loges2108) [reply src/pages/bracelets/Bracelates.tsx]: removed

File: src/interface/type.ts (hunk 6 above)
Comment (Loges2108) [reply src/interface/type.ts]: removed and moved to seperate file

File: src/common/component/commonCard/CommonProductCard.tsx (hunk 8 above)
Comment (Loges2108) [reply src/common/component/commonCard/CommonProductCard.tsx:9]: changed as IProduct

File: src/common/component/commonpages/CommonPage.tsx (hunk 2 above)
Comment (Loges2108) [reply src/common/component/commonpages/CommonPage.tsx]: change the all interface with prefix I

Comment (Loges2108) [reply src/common/component/commonpages/CommonPage.tsx]: solved

File: src/common/component/commonpages/CommonPage.tsx (hunk 5 above)
Comment (Loges2108) [reply src/common/component/commonpages/CommonPage.tsx]: working

File: src/common/component/commonpages/CommonPage.tsx (hunk 4 above)
Comment (Loges2108) [reply src/common/component/commonpages/CommonPage.tsx]: working

**Pull request merged**